dpx/src/dpx_pdfnames.rs

```diff
@@ -263,30 +262,20 @@ pub unsafe extern "C" fn pdf_names_close_object(
     0i32
 }
 #[inline]
-unsafe extern "C" fn cmp_key(mut d1: *const libc::c_void, mut d2: *const libc::c_void) -> i32 {
-    let mut cmp;
-    let sd1 = d1 as *const named_object;
-    let sd2 = d2 as *const named_object;
-    if (*sd1).key.is_null() {
-        cmp = -1i32
-    } else if (*sd2).key.is_null() {
-        cmp = 1i32
+fn cmp_key(sd1: &named_object, sd2: &named_object) -> ::core::cmp::Ordering {
+    use ::core::cmp::Ordering;
```

Comment:
1. This crate will never be no_std, so prefer using `std`
2. Fully qualified `::` prefix is unnecessary in all code except macros
3. If you’re using a type in a function signature as well, don’t put the `use` inside the function and then fully qualify it again in the signature.

So generally just put `use std::cmp::Ordering;` at the top of the file.